### ISSUE TITLE
Anyone started doing an RPM for Fedora with Wayland?

### ISSUE THREAD

**Issue by philiprhoades** [tenacityteam/tenacity-legacy#602]
### Guidelines

- [X] I have read the guidelines.

### Is your feature request related to a problem? Please describe.

There is no Tenacity RPM on the Fedora repos.

### Describe the solution you'd like

A Tenacity RPM on the Fedora repos - I can have a go at it if no-one else is interested . . although I haven't produced an RPM before . . 

### Describe alternatives you've considered

Continue with Audacity . .

### This issue is not a duplicate

- [X] I have confirmed this issue isn't a duplicate.

**Comment by rubyFeedback** [comment]
But isn't this up to the fedora maintainers? If tenacity compiles on linux then they can package it into a RPM right? I am not sure why the tenacity team should have to provide that for distribution maintainers. See the debian team radically repackaging stuff on their own; this used to cause problems for python and ruby (e. g. pip and gems split up).

**Comment by leio** [comment]
Fedora (or Red Hat at least) is a very big proponent of flatpak for desktop apps. So I think Fedora users should just use it from flatpak until tenacity has a release and someone packages it in RPM form for Fedora. And even then, flatpak is probably what they'd prefer for you to be using instead of that package, but then it's hopefully there in the future for those that prefer a system package.

Doing an official RPM package for Fedora is just too early when there's no Tenacity release yet.

**Comment by Be-ing** [comment]
AFAIK nobody has started. The first step would be [adding/updating Tenacity's dependencies](https://github.com/tenacityteam/tenacity/issues/90#issuecomment-901585714).

There are a handful of bugs with Wayland: #561, #562, #563. There is no pressing need to use Wayland rather than XWayland for Tenacity nor do I think there is a reason for packagers to force Tenacity to use Wayland.

**Comment by leio** [comment]
I don't think there's anything special to do to run with Wayland. It just does so. If we don't want to run it on Wayland, we actually would need to take steps to force X11/Xwayland. The flatpak we generate from CI is using XWayland just because the sandboxing gives access to X11 stuff, but not Wayland yet (and this was left like that due to these slight issues iirc).

**Comment by philiprhoades** [comment]
Thanks people - I will start with the Flatpak version and track developments after that - my preference would be for an RPM so maybe once things settle down a bit with dependencies etc, and if someone else hasn't already done an RPM, I will have another look at building it on Fedora and producing the RPM . .

Hmm . . just looked on FlatHub and Googled . . I can't find a Tenacity FlatPak . .

**Comment by Be-ing** [comment]
> once things settle down a bit with dependencies

What do you mean? If you want to package an RPM for Fedora, someone needs to update the dependencies in Fedora.

**Comment by philiprhoades** [comment]
> What do you mean?

I thought there was still some stuff on dependencies needing doing on the conversion from audacity?

Do you have a link for the FlatPak version?

**Comment by Be-ing** [comment]
> I thought there was still some stuff on dependencies needing doing

Yes, there is, as I [linked above](https://github.com/tenacityteam/tenacity/issues/90#issuecomment-901585714). If you take care of adding/updating the dependencies in Fedora, that would be great! I don't know what you mean by "settle down" though; AFAIK the dependencies don't need any more work.

> Do you have a link for the FlatPak version?

https://github.com/flathub/flathub/pull/2444

**Comment by philiprhoades** [comment]
Bad choice of words I guess - maybe "finish cleaning up fork"?

> If you take care of adding/updating the dependencies in Fedora, that would be great! 

I will try and look at it during the week . .

```
flatpak install --user https://dl.flathub.org/build-repo/56599/org.tenacityaudio.Tenacity.flatpakref
error: Can't load uri https://dl.flathub.org/build-repo/56599/org.tenacityaudio.Tenacity.flatpakref: Server returned status 404: Not Found
```

**Comment by nielsenb-jf** [comment]
The brave can try my [copr](https://copr.fedorainfracloud.org/coprs/nielsenb/tenacity/) which packages git revisions (at least until versioning settles down with Tenacity releases).

From a user standpoint, MIDI playback seems to not work, and it has the same UI hiccups as Audacity has under Wayland.

From a development standpoint, I also need to figure out how to handle RPATHs of the private libraries so building doesn't fail on F35+ with the `__brp_check_rpaths` check script enabled [which is now default](https://fedoraproject.org/wiki/Changes/Broken_RPATH_will_fail_rpmbuild).

**Comment by philiprhoades** [comment]
Thanks for the update! - I think I will look at this again when I update to F35.
